File: src/main/java/piles/SuitPile.java
package piles;

import deck.Card;
import deck.Rank;
import deck.Suit;

public class SuitPile extends Pile{
    private Suit suit;
    public SuitPile(Suit suit){
        this.suit=suit;
    }
    public boolean isCardValid(Card card) {
        boolean valid;
        if(getTopCard() != null){
            valid = card.getSuit() == this.suit &&
                    getTopCard().isPrevious(card);
        }else{
            valid = card.getRank() == Rank.ACE &&
                    card.getSuit() == this.suit;
        }
        return valid;
    }
}

File: src/main/scala/piles/SuitPile.scala
package piles

import deck.Card
import deck.Rank
import deck.Suit


class SuitPile(val suit: Suit) extends Pile {
  def isCardValid(card: Card): Boolean = {
    getTopCard match
      case Some(topCard) => (card.suit eq suit) && topCard.isPrevious(card)
      case None => (card.rank eq Rank.ACE) && (card.suit eq suit)
  }
}
